events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java

```diff
@@ -0,0 +1,263 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+package com.adobe.event.management;
+
+import com.adobe.event.management.model.EventMetadata;
+import com.adobe.event.management.model.Provider;
+import com.adobe.event.management.model.ProviderInputModel;
+import com.adobe.ims.JWTAuthInterceptor;
+import com.adobe.ims.util.TestUtil;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import feign.RequestInterceptor;
+import org.junit.Rule;
+import org.junit.Test;
+import java.util.*;
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertNotNull;
+
+public class ProviderServiceTest {
+
+  private static final String PROVIDER_ID = "619363ad-2461-4c1c-bb09-848f8b4d0a30";
+  private static final String SOURCE_ID = "urn:uuid:619363ad-2461-4c1c-bb09-848f8b4d0a30";
+
+  private static final String PROVIDER_MODEL_LABEL = "aio-lib-java Wiremock - test drive provider label";
+  private static final String PROVIDER_MODEL_DESCRIPTION = "aio-lib-java Wiremock - test drive provider description";
+  private static final String PROVIDER_MODEL_DOCS_URL = "https://localhost.com/test";
+
+  private static final String METADATA_EVENTCODE = "wiremock_eventcode";
+  private static final String METADATA_DESCRIPTION = "wiremock test description";
+  private static final String METADATA_LABEL = "test label";
+
+  @Rule
+  public WireMockRule wireMockRule = new WireMockRule(9999);
+
+  @Test
+  public void providerServiceTest() {
+
+    Optional<Provider> provider;
+    assertNotNull(getProviderService());
+
+    ProviderInputModel providerCreateModel = ProviderInputModel.builder()
+            .label(PROVIDER_MODEL_LABEL)
+            .description(PROVIDER_MODEL_DESCRIPTION)
+            .docsUrl(PROVIDER_MODEL_DOCS_URL)
+            .build();
+
+//    /** getProviders */
+    stubFor(get(urlEqualTo("/events/someConsumerOrgId/providers"))
+            .willReturn(okJson("{\"_links\":{\"self\":{\"href\":\"localhost:9999/events/someConsumerOrgId/providers\"}},\"_embedded\":{\"providers\":[{\"_links\":{\"rel:eventmetadata\":{\"href\":\"localhost:9999/events/providers/207aad29-f183-4c3d-87b5-22b21b8f0ee9/eventmetadata\"},\"self\":{\"href\":\"localhost:9999/events/providers/207aad29-f183-4c3d-87b5-22b21b8f0ee9\"}},\"id\":\"207aad29-f183-4c3d-87b5-22b21b8f0ee9\",\"label\":\"Analytics Triggers\",\"description\":\"Adobe Analytics Triggers events provider (gain insight into behavior of users on your site)\",\"source\":\"urn:uuid:207aad29-f183-4c3d-87b5-22b21b8f0ee9\",\"docs_url\":\"https://localhost.com/test\",\"publisher\":\"Adobe\"}]}}")
```

Comment:
the json payload should use the (wiremock port, provider* and event* ) constant too.
let's build a helper method may be

Reply:
I've created a few methods that are returning the responses which are being used more than once and adjusted them in order to use constants. Please let me know if you have another idea for this.